app/ide-desktop/lib/content/src/serviceWorker.ts

```diff
@@ -0,0 +1,32 @@
+/** @file A service worker that redirects paths without extensions to `/index.html`.
+ * This is only used in the cloud frontend. */
+/// <reference lib="WebWorker" />
+
+// =====================
+// === Fetch handler ===
+// =====================
+
+// We `declare` a variable here because Service Workers have a different global scope.
+// eslint-disable-next-line no-restricted-syntax
+declare const self: ServiceWorkerGlobalScope
+
+self.addEventListener('fetch', event => {
+    const url = new URL(event.request.url)
+    if (url.hostname === 'localhost' && url.pathname !== '/esbuild') {
+        event.respondWith(
+            fetch(event.request.url).then(response => {
+                const clonedResponse = new Response(response.body, response)
+                clonedResponse.headers.set('Cross-Origin-Embedder-Policy', 'require-corp')
+                clonedResponse.headers.set('Cross-Origin-Opener-Policy', 'same-origin')
+                clonedResponse.headers.set('Cross-Origin-Resource-Policy', 'same-origin')
+                return clonedResponse
+            })
+        )
```

Comment:
Also:
1. this is the same as code in other place - can we refactor it to one place maybe?
2. We have server built-into electron, why are we not using the same server with the watch scripts? would it not be cool to use the same code + make it possible to spawn electron in the watch mode? Ofc this is not part of this task, just a more general question.

Reply:
`./run ide watch` does use the same server, i believe, but `./run gui watch` doesn't spawn electron at all. it might be possible to spawn electron on `./run gui watch`, but in that case i'm not sure whether that would do anything different to `./run ide watch`?